components/content/inspira/ui/text-reveal-card/TextRevealCard.vue

```diff
@@ -0,0 +1,136 @@
+<template>
+  <div
+    @mouseenter="mouseEnterHandler"
+    @mouseleave="mouseLeaveHandler"
+    @mousemove="mouseMoveHandler"
+    @touchstart="mouseEnterHandler"
+    @touchend="mouseLeaveHandler"
+    @touchmove="touchMoveHandler"
+    ref="cardRef"
+    :class="[
+      'w-full max-w-[40rem] bg-[#1d1c20] border border-white/[0.08] rounded-lg p-4 sm:p-6 md:p-8 relative overflow-hidden',
+      props.class,
+    ]"
+  >
+    <h2 v-if="title" :class="['text-white text-lg mb-2', props.titleClass]">
+      {{ title }}
+    </h2>
+    <p
+      v-if="description"
+      :class="['text-[#a9a9a9] text-sm', props.descriptionClass]"
+    >
+      {{ description }}
+    </p>
+
+    <div class="h-40 relative flex items-center overflow-hidden">
+      <div
+        :style="{
+          width: '100%',
+          opacity: widthPercentage > 0 ? 1 : 0,
+          clipPath: `inset(0 ${100 - widthPercentage}% 0 0)`,
+          transition: isMouseOver ? 'none' : 'all 0.4s ease-out',
+        }"
+        class="absolute bg-[#1d1c20] z-20 will-change-transform"
+      >
+        <p
```

Comment:
Is it possible to use slots here also (for revealText and text)?
If it became possible, this will make this component more dynamic, as users can use text, or any div.

What do you think?

Reply:
I'll look into it. If the slot doesn’t break the styling, I'll use it.

Reply:
Let me know once that is done! Add base styling which is required, and rest can be added by the user.

If possible provide one more example!